darkstar_install_script.sh

```diff
@@ -0,0 +1,178 @@
+    ./configure CXXFLAGS=" -pthread" || exit
+  else
+    printf "\nRunning Default Install\n"
+    ./configure || exit
+  fi
+  make -j4 || exit
+  chown -R "${DARKSTAR_USER}":root /opt/darkstar/
+  printf "\nEnter the MySQL root password\nDefault: No password\n"
+  mysql -h "localhost" -u "root" -p -e "CREATE USER '${DARKSTAR_USER}'@'localhost' IDENTIFIED BY '${DARKSTAR_PASSWORD}';CREATE DATABASE dspdb;USE dspdb;GRANT ALL PRIVILEGES ON dspdb.* TO '${DARKSTAR_USER}'@'localhost';"
+  printf "\n"
+  cd ${INSTALL_DIR}/sql || exit
+  for SQL_FILE in *.sql
+    do
+      echo -n "Importing $SQL_FILE into the database..."
+      mysql dspdb -u "${DARKSTAR_USER}" "-p${DARKSTAR_PASSWORD}" < "$SQL_FILE" && echo "Success"
+    done
+  cd ..
+  # Systemd service files to be written
+  SYSTEMD_SERVICE_DSCONNECT="""
+[Unit]
+Description=Darkstar Final Fantasy XI - DS Connect
+Wants=network.target
+StartLimitIntervalSec=120
+StartLimitBurst=5
+
+[Service]
+Type=simple
+Restart=always
+RestartSec=5
+User=${DARKSTAR_USER}
+Group=${DARKSTAR_USER}
+WorkingDirectory=/opt/darkstar
+ExecStart=/opt/darkstar/dsconnect
+
+[Install]
+WantedBy=multi-user.target
+"""
+  SYSTEMD_SERVICE_DSGAME="""
+[Unit]
+Description=Darkstar Final Fantasy XI - DS Game
+Wants=network.target
+StartLimitIntervalSec=120
+StartLimitBurst=5
+
+[Service]
+Type=simple
+Restart=always
+RestartSec=5
+User=${DARKSTAR_USER}
+Group=${DARKSTAR_USER}
+WorkingDirectory=/opt/darkstar
+ExecStart=/opt/darkstar/dsgame
+
+[Install]
+WantedBy=multi-user.target
+"""
+  SYSTEMD_SERVICE_DSSEARCH="""
+[Unit]
+Description=Darkstar Final Fantasy XI - DS Search
+Wants=network.target
+StartLimitIntervalSec=120
+StartLimitBurst=5
+
+[Service]
+Type=simple
+Restart=always
+RestartSec=5
+User=${DARKSTAR_USER}
+Group=${DARKSTAR_USER}
+WorkingDirectory=/opt/darkstar
+ExecStart=/opt/darkstar/dssearch
+
+[Install]
+WantedBy=multi-user.target
+"""
+  printf "%s" "${SYSTEMD_SERVICE_DSCONNECT}" > /etc/systemd/system/darkstar-connect.service
+  printf "%s" "${SYSTEMD_SERVICE_DSGAME}" > /etc/systemd/system/darkstar-game.service
+  printf "%s" "${SYSTEMD_SERVICE_DSSEARCH}" > /etc/systemd/system/darkstar-search.service
+  chmod 755 /etc/systemd/system/darkstar*
+  systemctl daemon-reload
+  printf "\n\nInstallation Complete\n"
+fi
+chmod 775 /opt/darkstar -R 2>/dev/null
+echo
+if [ "${RUN_CHOICE}" == "1" ] || [ "${RUN_CHOICE}" == "2" ]; then
+  read -p "Would you like to update the Darkstar FFXI zone IP? (Y/N) " -n 1 -r ZONEIP
+  echo
+  if [[ ${ZONEIP} =~ ^[Yy]$ ]]; then
+    printf "\nYour current systems IPv4 Address are as follows\n"
+    ip -4 addr | grep -oP '(?<=inet\s)\d+(\.\d+){3}'
+    echo
+    read -r -p "Enter the IP address you wish to use with this Server: " DARKSTAR_IP
+    printf "\nEnter the MySQL root password\nDefault: No password\n"
+    mysql -h "localhost" -u "root" -p -e "USE dspdb;UPDATE zone_settings SET zoneip = '${DARKSTAR_IP}';"
+  fi
+  nano ${INSTALL_DIR}/conf/login_darkstar.conf
```

Comment:
Is the purpose of opening these files to allow the users to make changes?

Reply:
That is correct.  You have to set the newly created user and password for the SQL database access.

Reply:
I have now added text to explain why said files will be opened and how to use the text editor.